### PR TITLE
added light box support for datepicker

### PR DESCRIPTION
**Pull request by omerr132**

### CODE REVIEW

File: packages/aquery/src/utils/form/datePicker.js [hunk 1]
```diff
@@ -3,39 +3,72 @@ import randomId from '../randomId';
...
+        datePicker.setAttribute('format', format);
+
+        if (date.required) {
+            input.removeAttribute('readonly');
+            input.setAttribute('required', '');
+            datePicker.setAttribute('required', '');
+            console.log('addDatePicker', datePicker);
+        }
+        if (mode === "lightbox") {
+            const lightbox = document.createElement('amp-lightbox');
```
Comment (ofekrom) [inline packages/aquery/src/utils/form/datePicker.js:39]: From ln 38 to 63 should all go under the if else above. No need to check if light box again

**Review by ofekrom — CHANGES_REQUESTED** [review]

File: packages/aquery/src/utils/form/datePicker.js (hunk 1 above)
Comment (omerr132) [reply packages/aquery/src/utils/form/datePicker.js:39]: But there are  changes that are made to the datepicker before line 38. If I'll put this block before these changes (line 30), these changes would be missed. 
2nd reason is that there is an 'else' to this 'if'

Comment (ofekrom) [reply packages/aquery/src/utils/form/datePicker.js:39]: I don't understand your point.
I am looking at the code, and there is no reason to check if mode === "lightbox" twice

If you don't manage to do that, ping me tomorrow and we can do it together

Comment (omerr132) [reply packages/aquery/src/utils/form/datePicker.js:39]: OK I managed to do it.

File: packages/aquery/src/utils/form/datePicker.js [hunk 2]
```diff
@@ -4,28 +4,18 @@ const FORMATS = { FORMAT_US: 'MM-DD-YYYY', FORMAT_WORLD: 'DD-MM-YYYY' };
 
 const addDatePicker = (date, $) => {
     if (date) {
-        const mode = date.mode;
+        let mode_pre
+        date.mode ? mode_pre = date.mode : mode_pre = 'lightbox';
```
Comment (ofekrom) [inline packages/aquery/src/utils/form/datePicker.js:8]: You don't need mode_pre:
date.mode = date.mode ?  date.mode : 'lightbox';

File: packages/aquery/src/utils/form/datePicker.js [hunk 3]
```diff
@@ -3,39 +3,71 @@ import randomId from '../randomId';
...
+        }
+
+        if (date.mode && date.mode === 'static') {} else if (mode === "overlay") {
+            datePicker.setAttribute('mode', 'overlay');
+            datePicker.setAttribute('layout', 'container');
+            const parent = input.parentNode;
+            parent.appendChild(datePicker);
+            datePicker.appendChild(input);
+        } else {
+            var lightboxID = `lb${input.getAttribute('name')}`;
```
Comment (ofekrom) [inline packages/aquery/src/utils/form/datePicker.js:36]: Use const

Comment (omerr132) [reply packages/aquery/src/utils/form/datePicker.js:36]: but i need to use it outside the if. Will it work there if it's const?

Comment (ofekrom) [reply packages/aquery/src/utils/form/datePicker.js:36]: If you needed to use it outside of the scope of the if, var wouldn't help either...

You would have needed to declare the variable outside of the scope, and then initialize it inside the scope. If you needed to do that - it should have been "let" as using var is deprecated.

Anyhow, in this case it's a const.

**Review by ofekrom — APPROVED** [review]

**Pull request merged**